src/test/documentation_examples.rs

```diff
@@ -0,0 +1,1385 @@
+use futures::TryStreamExt;
+
+use crate::{
+    bson::{doc, Bson},
+    error::Result,
+    options::FindOptions,
+    test::TestClient,
+    Collection,
+};
+
+macro_rules! assert_coll_count {
```

Comment:
I originally wrote these as functions, but then I kept forgetting to `await` calls I made to them, so I instead just made them macros.